editor/src/clj/editor/editor_extensions.clj

```diff
@@ -463,45 +496,56 @@
                             msg     a string to output, might be multiline
     :save!                0-arg function that asynchronously saves any unsaved
                           changes, returns CompletableFuture (that might
-                          complete exceptionally if reload fails)"
-  [project kind & {:keys [reload-resources! display-output! save!] :as opts}]
+                          complete exceptionally if reload fails)
+    :open-resource!       1-arg function that asynchronously opens the supplied
+                          resource in an editor tab, returns CompletableFuture
+                          (that might complete exceptionally if resource could
+                          not be opened)"
+  [project kind & {:keys [reload-resources! display-output! save! open-resource!] :as opts}]
+  {:pre [reload-resources! display-output! save! open-resource!]}
   (g/with-auto-evaluation-context evaluation-context
     (let [extensions (g/node-value project :editor-extensions evaluation-context)
           old-state (ext-state project evaluation-context)
-          project-path (-> project
-                           (project/workspace evaluation-context)
-                           (workspace/project-path evaluation-context)
+          workspace (project/workspace project evaluation-context)
+          project-path (-> (workspace/project-path workspace evaluation-context)
                            .toPath
                            .normalize)
```

Comment:
Should it use `.toRealPath` instead of `.normalize`? To be honest, I kind of thing this is the wrong place to normalize the project path. 🤷